oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCheckCommand.java

```diff
@@ -242,16 +248,23 @@ static int checkDataStore(String... args) {
             }
 
             if (options.has(refOp) || options.has(consistencyOp)) {
-                if (options.has(verbose) &&
+
+                // Find blob ids by traversal for verbose mode + Segment Store or if verboseRootPath option
+                // is present (find blob references under a specific root path.)
+                if ((options.has(verbose) &&
                     (nodeStore instanceof SegmentNodeStore ||
-                        nodeStore instanceof org.apache.jackrabbit.oak.segment.SegmentNodeStore)) {
+                        nodeStore instanceof org.apache.jackrabbit.oak.segment.SegmentNodeStore)) ||
+                        options.has(verboseRootPath)) {
                     NodeTraverser traverser = new NodeTraverser(nodeStore, dsType);
                     closer.register(traverser);
-                    traverser.traverse();
+
+                    List<String> rootPathList = options.valuesOf(verboseRootPath);
+                    traverser.traverse((String[]) rootPathList.toArray(new String[rootPathList.size()]));
+
                     FileUtils.copyFile(traverser.references, register.createFile(refOp, dumpPath));
                 } else {
                     retrieveBlobReferences(blobStore, marker,
```

Comment:
So, as far as I understand, for SegmentStore, it just traverses the nodes, but for all other stores (MongoDB / DocumentStore), it gets all the references of all the nodes... Right? That would be a lot of references... I think it should traverse just the nodes in verboseRootPath in both cases, if verboseRootPath is set.

Reply:
Ah sorry, I see now! Looks good, I just didn't understand it.